auth/user_mgt_test.go

```diff
@@ -0,0 +1,716 @@
+					},
+					{
+						ProviderID:  "phone",
+						PhoneNumber: "+1234567890",
+					},
+				},
+				CustomClaims: map[string]interface{}{"admin": true, "package": "gold"},
+			},
+			PasswordHash: "passwordHash",
+			PasswordSalt: "passwordSalt",
+		},
+	}
+}
+
+func testCompareUserRecords(u1, u2 *UserRecord, t *testing.T) {
+	tests := []struct {
+		name string
+		got  interface{}
+		want interface{}
+	}{
+		{"user", u1, u2},
+		{"user.UID", u1.UID, u2.UID},
+		{"user.UserInfo", u1.UserInfo, u2.UserInfo},
+		{"user.ProviderUserInfo", u1.ProviderUserInfo, u2.ProviderUserInfo},
+		{"user.UserMetadata", u1.UserMetadata, u2.UserMetadata},
+	}
+	for k, pui := range u1.ProviderUserInfo {
+		tests = append(tests, struct {
+			name string
+			got  interface{}
+			want interface{}
+		}{fmt.Sprintf("Provider %d", k), pui, u2.ProviderUserInfo[k]})
+	}
+	for j, test := range tests {
+		if !reflect.DeepEqual(test.got, test.want) {
+			t.Errorf("test %d \n %s =  (%T) %#v \nwanted: (%T) %#v", j, test.name,
+				test.got, test.got, test.want, test.want)
+		}
+	}
+
+}
+
+// echoServer takes either a []byte or a string filename, or an object.
+//
+// echoServer returns a server whose client will reply with depending on the input type:
+//   * []byte: the []byte it got
+//   * string: the contents of the file named by the string in []byte form
+//   * object: the marshalled object, in []byte form
+//   * nil: "{}" empty json, in case we aren't interested in the returned value, just the marshalled request
+// The marshalled request is available through s.rbody, s being the retuned server.
+// It also returns a closing functions that has to be defer closed.
+func echoServer(resp interface{}, t *testing.T) (*mockAuthServer, func()) {
+	var b []byte
+	var err error
+	switch v := resp.(type) {
+	case nil:
+		b = []byte("")
+	case []byte:
+		b = v
+	case string:
+		fp := filepath.Join([]string{build.Default.GOPATH, "src", "firebase.google.com", "go", "testdata", v}...)
+		b, err = ioutil.ReadFile(fp)
+		if err != nil {
+			t.Fatal(err)
+		}
+	default:
+		b, err = json.Marshal(resp)
+		if err != nil {
+			t.Fatal("marshaling error")
+		}
+	}
+
+	s := mockAuthServer{Resp: b}
+
+	handler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		defer r.Body.Close()
+
+		reqBody, err := ioutil.ReadAll(r.Body)
+		if err != nil {
+			t.Error(err)
+		}
+		s.Req = append(s.Req, r)
+		s.rbody = reqBody
+		for k, v := range s.Header {
+			w.Header().Set(k, v)
+		}
+		if s.Status != 0 {
+			w.WriteHeader(s.Status)
+		}
+		w.Header().Set("Content-Type", "application/json")
+		w.Write(s.Resp)
+
+	})
+	s.srv = httptest.NewServer(handler)
+	conf := &internal.AuthConfig{
+		Creds: testCreds,
+		Opts: []option.ClientOption{
+			option.WithHTTPClient(s.srv.Client()),
+		},
+	}
+	authClient, err := NewClient(context.Background(), conf)
+	if err != nil {
+		t.Error()
+	}
+	authClient.url = s.srv.URL + "/"
+	s.Client = authClient
+	return &s, s.srv.Close
```

Comment:
Just expose a Close() function from mock server, that the caller can defer on. Then remove the 2nd return value.